src/controllers/QueryController.ts

```diff
@@ -580,16 +580,37 @@ export class QueryController extends RootComponent {
       return true;
     }
 
-    const enableHistory = this.searchInterface && this.searchInterface.options && this.searchInterface.options.enableHistory;
-    return enableHistory && this.getLastQueryHash() == this.queryHash(query, queryResults);
+    const enableHistory = !!(this.searchInterface && this.searchInterface.options && this.searchInterface.options.enableHistory);
+    return enableHistory && this.compareWithLastQueryHash(this.queryHash(query, queryResults));
+  }
+
+  private compareWithLastQueryHash(queryHash: string) {
+    const lastParams = JSON.parse(this.getLastQueryHash());
+    const newParams = JSON.parse(queryHash);
+    return Utils.objectEqual(lastParams, newParams);
   }
 
   private queryHash(query: IQuery, queryResults?: IQueryResults): string {
-    let queryHash = JSON.stringify(_.omit(query, 'firstResult', 'groupBy', 'debug'));
-    if (queryResults != null) {
-      queryHash += queryResults.pipeline;
+    const queryKeys: (keyof IQuery)[] = [
+      'q',
+      'aq',
+      'cq',
+      'dq',
+      'searchHub',
+      'tab',
+      'pipeline',
+      'sortCriteria',
+      'recommendation',
+      'commerce'
+    ];
+
+    const queryParams = pick(query, ...queryKeys);
+
+    if (queryResults) {
+      queryParams.pipeline = queryResults.pipeline;
```

Comment:
Why is this assignment needed?

Reply:
It was there before, I don't want to break & my guess is that sometimes the pipeline would just be defined in the response, not the request.